pkg/kubeapiserver/options/authentication.go

```diff
@@ -676,15 +677,18 @@ func (o *BuiltInAuthenticationOptions) ApplyTo(ctx context.Context, authInfo *ge
 					klog.ErrorS(err, "failed to validate authentication config")
 					// this config is not structurally valid and never will be, update the tracker so we stop retrying
 					trackedAuthenticationConfigData = authConfigData
+					authenticationconfigmetrics.RecordAuthenticationConfigAutomaticReloadFailure(apiServerID)
```

Comment:
Don't we need this for the load above too?  Maybe this should be done via a `defer` so it is easier to track?

Reply:
ha, I suggested it not be a defer so it was obvious all the places we trip a failure metric ... I only care about 3 out of 10 but I found the defer harder to reason about (as well as counting on `err` not being masked through the whole function)